mass_mailing_switzerland/models/crm_event_compassion.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Nathan Fluckiger <nathan.fluckiger@hotmail.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import models, api
+
+
+class EventCompassion(models.Model):
+
+    _inherit = 'crm.event.compassion'
+
+    @api.model
+    def create(self, vals):
+        event = super(EventCompassion, self).create(vals)
+        for NewEvent in self:
+            if NewEvent.campaign_id is not None:
```

Comment:
`if event.campaign_id` is enough for the test. Relational fields are equal to False in case they are not set.